pennylane/devices/task_qubit.py

```diff
@@ -0,0 +1,246 @@
+        "Defines a serializer for the Dask backend"
+        header, frames = dist.protocol.numpy.serialize_numpy_ndarray(tensor)
+        header["type"] = "qml.numpy.tensor"
+        header["requires_grad"] = tensor.requires_grad
+        frames = [tensor.data]
+        return header, frames
+
+    @dask_deserialize.register(qml.numpy.tensor)
+    def deserialize(header: Dict, frames: List[bytes]) -> qml.numpy.tensor:
+        "Defines a deserializer for the Dask backend"
+        return qml.numpy.tensor(frames[0], requires_grad=header["requires_grad"])
+
+
+except ImportError as e:  # pragma: no cover
+    raise ImportError("task.qubit requires installing dask and dask.distributed") from e
+
+
+class ProxyInstanceCLS(classmethod):
+    """
+    This utility class allows the use of both an instance
+    as well as class method types. For situations where
+    the explicit class method is needed, we can use this as a
+    decorator, and select the appropriate call dynamically.
+    This is an essential support for accessing backend
+    functionality supports with the proxy `task.qubit`.
+
+    The implementation is based on the Python descriptor guide
+    as mentioned https://stackoverflow.com/questions/28237955/same-name-for-classmethod-and-instancemethod/28238047#28238047
+    """
+
+    def __get__(self, instance, type_):
+        descr_get = super().__get__ if instance is None else self.__func__.__get__
+        return descr_get(instance, type_)
+
+
+# pylint: too-few-public-methods, bad-classmethod-argument
+class TaskQubit(QubitDevice):
+    """Proxy simulator plugin written using Dask.Distributed as a task-distribution scheduling backend.
+
+    **Short name:** ``task.qubit``
+
+    This device provides a pure-state qubit simulator wrapping both ``"default.qubit"`` and ``"lightning.qubit"``,
+    and written to allow batched offloading to a Dask scheduler.
+
+    To use this device, you will need to install dask and dask.distributed:
+
+    .. code-block:: console
+
+        pip install dask distributed
+
+    **Example**
+
+    >>> import pennylane as qml
+    >>> import tensorflow as tf
+    >>> import dask.distributed as dist
+
+    >>> if __name__ == '__main__': # Required for LocalCluster
+    ...     cluster = dist.LocalCluster(n_workers=4, threads_per_worker=1)
+    ...     client = dist.Client(cluster)
+    ...     backend = "default.qubit"
+    ...     dev = qml.device("task.qubit", wires=6, backend=backend)
+    ...     @qml.qnode(dev, cache=False, interface="tf", diff_method="parameter-shift) # caching must be disabled due to proxy interface
+    ...     def circuit(x):
+    ...         qml.RX(x[0], wires=0)
+    ...         qml.RY(x[1], wires=0)
+    ...         qml.RZ(x[2], wires=0)
+    ...         qml.RZ(x[0], wires=1)
+    ...         qml.RX(x[1], wires=1)
+    ...         qml.RY(x[2], wires=1)
+    ...         return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+    ...     weights = tf.Variable(qml.numpy.random.rand(3))
```

Comment:
Nitpick: we could have an import for `numpy` at the beginning and have `np.random.rand(3)` here to get users into the habit of import numpy from PennyLane. :smile:

Reply:
Sure thing!